### DISCUSSION TITLE
Views sometimes come with drawbacks, so are Domain Representations supported?

### DISCUSSION THREAD

**Discussion by thadguidry** [9tigerio/db2rest#208] [Ideas]
See: [Domain Representations](https://postgrest.org/en/stable/references/api/domain_representations.html)
> Domain Representations separates “how the data is presented” from “how the data is stored”. It works by creating [domains](https://www.postgresql.org/docs/current/sql-createdomain.html) and [casts](https://www.postgresql.org/docs/current/sql-createcast.html), the latter act on the former to present and receive the data in different formats.

Views come with drawbacks that do increase complexity.
For details see [Advantages over Views](https://postgrest.org/en/stable/references/api/domain_representations.html#advantages-over-views)

> Domain Representations avoid all the above drawbacks. Their only drawback is that for existing tables, you have to change the column types. But this should be a fast operation since domains are binary coercible with their underlying types. A table rewrite won’t be required.

**Comment by kdhrubo** [comment]
I believe this can be supported in both a database native way as well as a database-independent way. 

For example, DB2Rest has plans for such representation in the roadmap. 
https://sqids.org/
